thirdwavelist/android
# Bump to latest Gradle and add more Security features

## Changes

Bump to latest Gradle and use Security features

- Updates to [7.5-rc01](https://github.com/gradle/gradle/releases/tag/v7.5.0-RC1)
- Removed unnecessary gradle cache restoration on workflows that do not use cache at all
- Enabled checksum and signature validation by Gradle
- Add checksum validation for gradle wrapper
- Add archive build reports step to build workflow
- Newlines in GHA workflow between steps for better readability

To update/generate new checksum and signature metadata, all you need to do is run:
```
./gradlew --write-verification-metadata sha256,pgp
```

Then check in `./gradle/verification-metadata.xml`

See:
- [Main inspiration for this](https://www.liutikas.net/2022/05/13/Gradle-Security-Considerations.html)
- [Gradle verifying dependencies](https://docs.gradle.org/current/userguide/dependency_verification.html)
- [Gradle wrapper validator](https://github.com/marketplace/actions/gradle-wrapper-validation)
- [Gradle stable distributions](https://services.gradle.org/distributions/)
- [Gradle checksums for stable releases](https://gradle.org/release-checksums/)

Signed-off-by: Antal János Monori <anthonymonori@gmail.com>

## Checklist
- [n/a] Appropriate test coverage was added
